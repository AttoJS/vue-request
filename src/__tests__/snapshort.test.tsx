import { mount } from '@vue/test-utils';
import { RequestConfig } from '../index';
import { defineComponent } from 'vue';

describe('RequestConfig', () => {
  const Child = defineComponent({
    setup() {
      return () => <button>button</button>;
    },
  });
  it('RequestConfig default slots should work ', () => {
    const wrapperA = mount(
      defineComponent({
        setup() {
          return () => (
            <RequestConfig config={{ loadingDelay: 0 }}>
              this is a text
            </RequestConfig>
          );
        },
      }),
    );
    expect(wrapperA.html()).toMatchSnapshot();
    wrapperA.unmount();

    const wrapperB = mount(
      defineComponent({
        setup() {
          return () => (
            <RequestConfig config={{ loadingDelay: 0 }}>
              this is a text
              <Child />
            </RequestConfig>
          );
        },
      }),
    );
    expect(wrapperB.html()).toMatchSnapshot();
  });
});
