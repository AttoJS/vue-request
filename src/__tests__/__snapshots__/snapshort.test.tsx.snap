// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestConfig RequestConfig default slots should work  1`] = `"this is a text"`;

exports[`RequestConfig RequestConfig default slots should work  2`] = `"this is a text<button>button</button>"`;
